### PR TITLE
feat(mute): mute everyone / everyone else

### PR DESCRIPTION
**Pull request by gimre**

### CODE REVIEW

**Comment by damencho** [comment]
Jenkins, add to whitelist.

File: lang/main.json [hunk 1]
```diff
@@ -213,6 +213,11 @@
         "muteParticipantButton": "Mute",
         "muteParticipantDialog": "Are you sure you want to mute this participant? You won't be able to unmute them, but they can unmute themselves at any time.",
         "muteParticipantTitle": "Mute this participant?",
+        "muteMultipleParticipantBody": "You won't be able to unmute them, but they can unmute themselves at any time.",
+        "muteMultipleParticipantButton": "Mute",
```
Comment (damencho) [inline lang/main.json]: Please do not duplicate already available strings. When reusing the already available keys if appropriate, you reuse more than 30 languages translations.

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    _onSubmit() {
+        const {
+            dispatch,
+            exclude,
+            localParticipantId,
+            remoteParticipantIds
+        } = this.props;
+
+        remoteParticipantIds
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: Let's move the whole mute logic to a separate action. It will probably need to be async action to select the participants.

This way you won't need to `mapStateToProps` at all and the dialog wouldn't be rerendered every time when something changes in the participants.

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+/**
+ * Maps part of the redux state to the component's props.
+ *
+ * @param {Object} state - The redux store/state.
+ * @returns {Object}
+ */
+function _mapStateToProps(state: Object) {
+    return {
+        localParticipantId: state['features/base/participants']
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: You can use `getLocalParticipant` from base/participants to select the local participant.

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    render() {
+        const { participantID, t } = this.props;
+
+        return (
+            <RemoteVideoMenuButton
+                buttonText = { t('videothumbnail.domuteOthers') }
+                displayClass = { 'mutelink' }
+                icon = { IconMicDisabled }
+                id = { `mutelink_${participantID}` }
+                // eslint-disable-next-line react/jsx-handler-names
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: Why?

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+import AbstractMuteButton, {
+    _mapStateToProps,
+    type Props
+} from '../AbstractMuteButton';
+
+import RemoteVideoMenuButton from './RemoteVideoMenuButton';
+import MuteEveryoneDialog from '../../../remote-video-menu/components/web/MuteEveryoneDialog';
+
+/**
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: This JSDoc is copy/paste from somewhere else and it doesn't explain anything about the component.

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import { IconMicDisabled } from '../../../base/icons';
+import { connect } from '../../../base/redux';
+
+import AbstractMuteButton, {
+    _mapStateToProps,
+    type Props
+} from '../AbstractMuteButton';
+
+import RemoteVideoMenuButton from './RemoteVideoMenuButton';
+import MuteEveryoneDialog from '../../../remote-video-menu/components/web/MuteEveryoneDialog';
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: Please arrange the imports according to https://github.com/jitsi/jitsi-meet/blob/master/doc/coding-style.md#formatting

`../AbstractMuteButton` and `./RemoteVideoMenuButton` are from the same feature, they need to be grouped last. And `'../../../remote-video-menu/components/web/MuteEveryoneDialog'` goes to the second group.

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * NOTE: At the time of writing this is a button that doesn't use the
+ * {@code AbstractButton} base component, but is inherited from the same
+ * super class ({@code AbstractMuteButton} that extends {@code AbstractButton})
+ * for the sake of code sharing between web and mobile. Once web uses the
+ * {@code AbstractButton} base component, this can be fully removed.
+ */
+class MuteEveryoneElseButton extends AbstractMuteButton {
+    /**
+     * Instantiates a new {@code Component}.
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: Maybe at least:
`Instantiates a new {@code MuteEveryoneElseButton}`

File: react/features/toolbox/components/MuteEveryoneButton.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    _downloadAppsUrl: string,
+
+    /**
+     * The ID of the local participant.
+    */
+    localParticipantId: string
+};
+
+/**
+ * Implements an {@link AbstractButton} to open the applications page in a new window.
```
Comment (hristoterezov) [inline react/features/toolbox/components/MuteEveryoneButton.js]: Really?

File: react/features/toolbox/components/MuteEveryoneButton.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * The URL to the applications page.
+     */
+    _downloadAppsUrl: string,
```
Comment (hristoterezov) [inline react/features/toolbox/components/MuteEveryoneButton.js]: This is not used

File: react/features/toolbox/components/MuteEveryoneButton.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * Maps part of the redux state to the component's props.
+ *
+ * @param {Object} state - The redux store/state.
+ * @returns {Object}
+ */
+function _mapStateToProps(state: Object) {
+    return {
+        localParticipantId: state['features/base/participants']
```
Comment (hristoterezov) [inline react/features/toolbox/components/MuteEveryoneButton.js]: Again please use `getLocalParticipant`

File: react/features/toolbox/components/web/Toolbox.js [hunk 11]
```diff
@@ -1336,6 +1348,10 @@ function _mapStateToProps(state) {
         _hideInviteButton:
             iAmRecorder || (!addPeopleEnabled && !dialOutEnabled),
         _isGuest: state['features/base/jwt'].isGuest,
+        _isModerator: state['features/base/participants']
```
Comment (hristoterezov) [inline react/features/toolbox/components/web/Toolbox.js]: Please use `getLocalParticipant`

File: react/features/toolbox/components/web/Toolbox.js [hunk 12]
```diff
@@ -996,6 +1004,10 @@ class Toolbox extends Component<Props, State> {
                 key = 'settings'
                 showLabel = { true }
                 visible = { this._shouldShowButton('settings') } />,
+            <MuteEveryoneButton
+                key = 'mute-everyone'
+                showLabel = { true }
+                visible = { _isModerator } />,
```
Comment (hristoterezov) [inline react/features/toolbox/components/web/Toolbox.js]: Why this button is not controlled by the interface_config?

Comment (hristoterezov) [inline react/features/toolbox/components/web/Toolbox.js]: Also I'm wondering if the `_isModerator` flag should be moved to `MuteEveryoneButton`'s `mapStateToProps`?

File: react/features/toolbox/components/MuteEveryoneButton.js [hunk 13]
```diff
@@ -0,0 +1,69 @@
+// @flow
+
+import { createToolbarEvent, sendAnalytics } from '../../analytics';
+import { openDialog } from '../../base/dialog';
+import { translate } from '../../base/i18n';
+import { IconMicDisabled } from '../../base/icons';
+import { connect } from '../../base/redux';
+import { AbstractButton, type AbstractButtonProps } from '../../base/toolbox';
+import MuteEveryoneDialog from '../../remote-video-menu/components/web/MuteEveryoneDialog';
```
Comment (hristoterezov) [inline react/features/toolbox/components/MuteEveryoneButton.js]: Why not import directly from the feature?

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 6 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: Also why not import directly the feature?

File: react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js [hunk 14]
```diff
@@ -8,11 +8,12 @@ import {
 } from '../../analytics';
 import { muteRemoteParticipant } from '../../base/participants';
 
+/* eslint-disable no-unused-vars */
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js]: Why?

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+            dispatch,
+            exclude,
+            localParticipantId,
+            remoteParticipantIds
+        } = this.props;
+
+        remoteParticipantIds
+            .filter(id => !exclude.includes(id))
+            .forEach(id => {
+                if (id === localParticipantId) {
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: When you are looping trough the remote participants probably you won't have the local participant id there at all :)

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+                    const audioMuted = true;
+
+                    sendAnalytics(createToolbarEvent(AUDIO_MUTE, { enable: audioMuted }));
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: This whole thing is totally copy pasted from the audio mute button in the toolbar....

I guess you can find better solution.

**Comment by gimre** [comment]
Thank you for your thorough review. A few things that I feel I need to mention:

- the implementation is tested and works as intended
- comments might indeed be not matching with the actual code since I used components that were already there as a starting point in hopes of an easier merge process - didn't work out
- all eslint ignores were already there to begin with in other parts of the repo
- order of imports and other arbitrary things might not be apparent to a first committer

I think this covers >90% of the comments that you had for this PR.

I think we can do better at respecting each other's work since we are all working towards the same goal in the end.

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 16 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: I am doing invidual mutes in a loop - what is the problem with sending analytics individually?

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 15 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: The local participant was there while testing..

File: lang/main.json (hunk 1 above)
Comment (gimre) [reply lang/main.json]: Thanks - my intent was to keep texts separate in case they need to reworded later. Will change to reuse existing strings.

File: react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js (hunk 14 above)
Comment (gimre) [reply react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js]: eslint and flow were exclusively not seeing the APP global being used (when one was satisfied, the other was not) - was the only solution I could find in the end

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 3 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: Didn't know about that - will change, thanks

File: react/features/toolbox/components/MuteEveryoneButton.js (hunk 9 above)
Comment (gimre) [reply react/features/toolbox/components/MuteEveryoneButton.js]: Will remove - thanks

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 15 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: OK. I see. Than please name your props better!

**Comment by hristoterezov** [comment]
I'm sorry if I offended you, that was not the intention! Please see my responses inline:

> Thank you for your thorough review. A few things that I feel I need to mention:
> 
> * the implementation is tested and works as intended

My bad! The problems turn out to be the naming of the props and I got confused. Sorry!

> * comments might indeed be not matching with the actual code since I used components that were already there as a starting point in hopes of an easier merge process - didn't work out

Not a big deal! But in future please read your PR before creating it. I mean everybody can miss something like this but it is basically one additional review/change cycle and it is easy to be prevented!

> * all eslint ignores were already there to begin with in other parts of the repo

So this means that we should keep adding more? I don't understand your point here.

Explanation why do we have eslint ignores:
Since this is relatively old project we have a lot of code which was written before we started using ESLint. When we started using ESLint we needed to add some ignores here and there and we gradually started to update our code and fix those issues. Unfortunately as you can see not everything is fixed yet but I hope one day we will. 

> * order of imports and other arbitrary things might not be apparent to a first committer

Sure! That's why I've sent you a link to our code styling document. And that's why we have mandatory reviews.

If you have ideas how to improve the situation for first committer please share them!

File: react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js (hunk 14 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js]: How about `declare var APP: Object;`?

**Comment by hristoterezov** [comment]
> I think this covers >90% of the comments that you had for this PR.

Actually you haven't replied to many comments! Please do!

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 16 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: There isn't a problem with sending individual analytics. The problem is that  you've copy pasted relatively big chunk of code from other parts of the project. The least you can do is to create a function which will be used in both places. WDYT?

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 15 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: OK. Than obviously we have a naming problem here! Please fix this!

Also we have the `.local` prop for every participant.

File: react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js [hunk 17]
```diff
@@ -35,15 +36,15 @@ type Props = {
  *
  * @extends Component
  */
-export default class AbstractMuteRemoteParticipantDialog
-    extends Component<Props> {
+export default class AbstractMuteRemoteParticipantDialog<P:Props = Props>
```
Comment (saghul) [inline react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js]: Why is this change necessary?

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 18]
```diff
@@ -0,0 +1,137 @@
+/* @flow */
```
Comment (saghul) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: Please use `// @flow` instead.

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+
+    /**
+     * The IDs of the remote participants to be muted.
+     */
+    remoteParticipantIds: Array<string>,
+
+     /**
+     * The IDs of the remote participants to exclude from being muted.
+     */
+    exclude: Array<string>,
```
Comment (saghul) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: style: please sort alphabetically

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+                    sendAnalytics(createToolbarEvent(AUDIO_MUTE, { enable: audioMuted }));
+                    dispatch(setAudioMuted(audioMuted, true));
+
+                    // FIXME: The old conference logic as well as the shared video feature
+                    // still rely on this event being emitted.
+                    typeof APP === 'undefined'
+                        || APP.UI.emitEvent(UIEvents.AUDIO_MUTED, audioMuted, true);
+                } else {
+                    const props = {
+                        dispatch,
```
Comment (saghul) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: why do you need to pass diapcth around?

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+            .forEach(id => {
+                if (id === localParticipantId) {
+                    const audioMuted = true;
+
+                    sendAnalytics(createToolbarEvent(AUDIO_MUTE, { enable: audioMuted }));
+                    dispatch(setAudioMuted(audioMuted, true));
+
+                    // FIXME: The old conference logic as well as the shared video feature
+                    // still rely on this event being emitted.
+                    typeof APP === 'undefined'
```
Comment (saghul) [inline react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: You are on a web component, you don't need to check for this.

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js [hunk 22]
```diff
@@ -0,0 +1,77 @@
+/* @flow */
```
Comment (saghul) [inline react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: same as before

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 6 above)
Comment (saghul) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: That is prone to intoducing cycles, importing the dialog directly is not a bad idea.

File: react/features/remote-video-menu/components/web/RemoteVideoMenuTriggerButton.js [hunk 23]
```diff
@@ -15,6 +15,7 @@ import {
     RemoteVideoMenu,
     VolumeSlider
 } from './';
+import MuteEveryoneElseButton from './MuteEveryoneElseButton';
```
Comment (saghul) [inline react/features/remote-video-menu/components/web/RemoteVideoMenuTriggerButton.js]: why not import this button like the others above?

File: react/features/toolbox/components/MuteEveryoneButton.js (hunk 13 above)
Comment (saghul) [reply react/features/toolbox/components/MuteEveryoneButton.js]: @hristoterezov Same answer as before.
@gimre Note that this button is not on web/ so it should be used on mobile, and then you are importing a web dialog, which is a no-no.

File: react/features/toolbox/components/web/Toolbox.js (hunk 12 above)
Comment (saghul) [reply react/features/toolbox/components/web/Toolbox.js]: Makes sense indeed!

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 6 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: For now we have this practice only if there isn't any other obvious solution. Do we have this problem here? If not we should definitely import the feature, at least for the sake of consistency.

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 4 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: So, I took the example from:
react/features/remote-video-menu/components/web/MuteButton.js:60

After some further investigation it seems this is necessary because in AbstractMuteButton the click handler is named without 'on' thus breaking the 'react/jsx-handler-names' rule.

How should I proceed then?

File: react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js (hunk 17 above)
Comment (gimre) [reply react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js]: Needed to extend the abstract Props provided by `components/AbstractMuteRemoteParticipantDialog` - other uses didn't need this - it's similar to what is being done in `react/features/base/toolbox/components/AbstractButton.js` but with a default to abstract Props so there is no change needed in other places

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 21 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: Thank you, did not know this.

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 20 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: because `AbstractMuteRemoteParticipantDialog`'s `_onSubmit` expects it. do you think doing super._onSubmit.call(this) is better? I can do that instead if you feel it's better

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 18 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: thanks! changing. (fun fact: there are 104 instances of using `/* @flow */ in other places in the code :D)`

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 15 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: > OK. Than obviously we have a naming problem here! Please fix this!
> 
> Also we have the `.local` prop for every participant.

renamed the prop. not sure what you mean by this

File: react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js (hunk 14 above)
Comment (gimre) [reply react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js]: after further checking not sure how this got here - removed.

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 22 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: changed

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 5 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: updated

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 7 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: example lifted from `react/features/remote-video-menu/components/web/MuteButton.js:28`

Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: updated

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 19 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: changed

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 21 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: changed

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 20 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: actually can't use `this` because I need to use a different `participantID` on each call

Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: for context, I took this approach to reuse as much as I could from existing functionality

File: react/features/toolbox/components/web/Toolbox.js (hunk 12 above)
Comment (gimre) [reply react/features/toolbox/components/web/Toolbox.js]: I looked into this, it seems this toolbox item is the only one that needs to be visible if you are a moderator. There is quite a complicated inheritance chain behind all toolbox items - do you have any suggestion on how to move this visibility flag into `MuteEveryoneButton` without needing to alter a lot of other components along and across the inheritance chain?

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 4 above)
Comment (saghul) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: I'd say you can leave it, as this is the case already with MuteButton.

File: react/features/toolbox/components/web/Toolbox.js (hunk 12 above)
Comment (saghul) [reply react/features/toolbox/components/web/Toolbox.js]: I think a pattern like this would work: https://github.com/jitsi/jitsi-meet/blob/98698ba89accac8fa707095cbc5020199012686c/react/features/etherpad/components/SharedDocumentButton.js#L73

Comment (gimre) [reply react/features/toolbox/components/web/Toolbox.js]: updated

Comment (gimre) [reply react/features/toolbox/components/web/Toolbox.js]: thank you @saghul , I missed that :( updated

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 20 above)
Comment (saghul) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: ack, leave it then

Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: alrighty, ty

File: react/features/remote-video-menu/components/web/RemoteVideoMenuTriggerButton.js (hunk 23 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/RemoteVideoMenuTriggerButton.js]: ty!

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 2 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: Good point - updated.

File: react/features/remote-video-menu/actions.js [hunk 24]
```diff
@@ -12,3 +27,56 @@ import { RemoteVideoMenu } from './components';
...
+        const state = getState();
+        const localId = getLocalParticipant(state).id;
+        const participantIds = state['features/base/participants']
+            .map(p => p.id);
+
+        /* eslint-disable no-confusing-arrow */
+        participantIds
+            .filter(id => !exclude.includes(id))
+            .map(id => id === localId ? muteLocal() : muteRemote(id))
+            .map(dispatch);
```
Comment (hristoterezov) [inline react/features/remote-video-menu/actions.js]: So in order to dispatch the everything we practically go over the participant array 4 times. Wouldn't this be better:
```
state['features/base/participants'].forEach(({ id }) => {
    if (!exclude.includes(id)) {
        dispatch(id === localId ? muteLocal() : muteRemote(id));
    }
});
```

IMHO looks simpler and goes over the participants array only ones. WDYT?

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 6 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: @gimre Could you please either change the import to use the feature or either add a comment why this is problematic?

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+import { openDialog } from '../../../base/dialog';
+import { translate } from '../../../base/i18n';
+import { IconMicDisabled } from '../../../base/icons';
+import { connect } from '../../../base/redux';
+
+import AbstractMuteButton, {
+    _mapStateToProps,
+    type Props
+} from '../AbstractMuteButton';
+
```
Comment (hristoterezov) [inline react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: style: redundant empty line

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 6 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: but wait actually this is the same exact directory, isn't it? Why not import `./MuteEveryoneDialog`?

File: react/features/toolbox/components/MuteEveryoneButton.js (hunk 13 above)
Comment (hristoterezov) [reply react/features/toolbox/components/MuteEveryoneButton.js]: @gimre Could you please either change the import to use the feature or either add a comment why this is problematic?

Comment (hristoterezov) [reply react/features/toolbox/components/MuteEveryoneButton.js]: Also please address @saghul comment. I guess you can move the component to the web folder since it is anyway imported only by the web toolbox.

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 15 above)
Comment (hristoterezov) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: I meant that while looping trough the participant you can check if the participant is local or not by checking the `.local` property. This doesn't matter now.

File: react/features/remote-video-menu/actions.js [hunk 26]
```diff
@@ -12,3 +27,56 @@ import { RemoteVideoMenu } from './components';
 export function hideRemoteVideoMenu() {
     return hideDialog(RemoteVideoMenu);
 }
+
+/**
+ * Mutes the local participant.
+ *
+ * @returns {Function}
+ */
+export function muteLocal() {
```
Comment (hristoterezov) [inline react/features/remote-video-menu/actions.js]: Since this is very similar to what we do in AudioMuteButton.js, I'm wondering can't we start dispatching it there too?

File: react/features/remote-video-menu/actions.js (hunk 24 above)
Comment (gimre) [reply react/features/remote-video-menu/actions.js]: I can change it to that if you like that more - but as a general idea, (1 iteration × 4 operations) === (4 iterations x 1 operation). There are more things going on here, but generally speaking.

Comment (hristoterezov) [reply react/features/remote-video-menu/actions.js]: Actually good point! Now I agree it is probably not a big deal. It is your choice than! Whatever you prefer, just let me know!

File: react/features/remote-video-menu/actions.js (hunk 26 above)
Comment (gimre) [reply react/features/remote-video-menu/actions.js]: Updated AudioMuteButton.js to this new action!

**Comment by gimre** [comment]
@hristoterezov 
Thanks for the effort reviewing - I initially opened this PR as a work in progress since I was expecting a lot of comments, I could have communicated that better in the beggining. Water under the bridge.

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 25 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: removed

File: react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js (hunk 6 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneElseButton.js]: changed to import sibling './MuteEveryoneDialog'

File: react/features/toolbox/components/MuteEveryoneButton.js (hunk 13 above)
Comment (gimre) [reply react/features/toolbox/components/MuteEveryoneButton.js]: moved it to web/

File: react/features/remote-video-menu/components/web/MuteEveryoneDialog.js (hunk 16 above)
Comment (gimre) [reply react/features/remote-video-menu/components/web/MuteEveryoneDialog.js]: i guess since this is now consolidated in an action, this is also resolved

File: react/features/remote-video-menu/actions.js (hunk 24 above)
Comment (gimre) [reply react/features/remote-video-menu/actions.js]: I'll leave it as is if its ok with you!

**Comment by gimre** [comment]
@hristoterezov all comments addressed

**Comment by hristoterezov** [comment]
> @hristoterezov
> Thanks for the effort reviewing - I initially opened this PR as a work in progress since I was expecting a lot of comments, I could have communicated that better in the beginning. Water under the bridge.

@gimre If I knew, I wouldn't be so picky about the JSDocs, naming and code style. I'm sorry if you indicated somehow that the PR is draft or WIP and I missed it somehow!
